docker/service/src/main/java/uk/nhs/adaptors/scr/controllers/validation/getscrid/type/TypeCodeParameterValidator.java

```diff
@@ -0,0 +1,46 @@
+package uk.nhs.adaptors.scr.controllers.validation.getscrid.type;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.scr.exceptions.BadRequestException;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TypeCodeParameterValidator implements ConstraintValidator<TypeCodeParameter, String> {
+
+    private static final String SUPPORTED_TYPE = "http://snomed.info/sct|196981000000101";
+
+    @Override
+    public void initialize(TypeCodeParameter contactNumber) {
+    }
+
+    @Override
+    public boolean isValid(String type, ConstraintValidatorContext context) {
+        try {
+            checkGetScrTypeParam(type);
+        } catch (BadRequestException exc) {
+            setErrorMessage(context, exc.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private void checkGetScrTypeParam(String type) {
```

Comment:
I don't think there's a point in exception-driven approach here. It's just one check - why not return boolean?
This applies to all new validators.

Reply:
Changed validation checks to return booleans.